core/src/com/unciv/models/metadata/GameSettings.kt

```diff
@@ -40,6 +43,11 @@ class GameSettings {
     }
 
     fun save(){
+        if (Gdx.app.type == Application.ApplicationType.Desktop) {
+            val gm = Gdx.graphics.displayMode
+            val gr = Gdx.graphics
```

Comment:
Since you mentioned these lines, I don't know why do you call it reflection. I understand a "reflection" as RTTI (Run-time type information). You do not work with the type information here, you just get the "pointer" to the object.
Also, beside the strange naming, I would reuse `gr` for getting `gm`, does it make sense?

Reply:
> why do you call it reflection
I may be mistaken in word choice. But - just set a breakpoint at val gr = and look at what the debugger says about gr, and what the auto-annotations said before, then use its object view: A widely different set of members than you get from typing "gr.".
IMHO reflection gets a "pointer" then uses a reflection API to query class and member information and/or call methods using strings for all names - and has access to private and otherwise hidden internals. (gr as LwjglGraphics).config.x (inacessible) becomes ImaginaryReflectionApi.GetPropertyValue(pointer = gr, classname = "LwjglGraphics", membername = "x", type = java.Int) or wildly unsimilar. I vaguely remember using it in c# - but there you can infer most details from debugger inspection windows, and there was ReSharper and lots of examples.